### PR TITLE
Add build instructions for Linux, Windows, Mac

### PR DESCRIPTION
**Pull request by Deniskore**

### CODE REVIEW

**Comment by Deniskore** [comment]
@skywinder I don't mind.

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+cmake -G "Ninja" ..
+ninja
+```
+
+## Building on Mac
+
+1. Install Xcode developer tools and [vcpkg](https://github.com/Microsoft/vcpkg)
+2. Install all relevant dependencies:
+```
+vcpkg install openssl zlib getopt libmicrohttpd
```
Comment (salemgolemugoo) [inline BUILD.md:70]: Why do you use vcpkg together with brew? Wouldn't it be better to use brew only?

File: BUILD.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+```
+3. Install ninja and cmake
+```
+brew install ninja
+brew install cmake
```
Comment (salemgolemugoo) [inline BUILD.md:75]:
```suggestion
brew install cmake
brew install openssl
```

File: BUILD.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+```
+4. Compile
+```
+cd telegram-node
+mkdir build
+cd build
+cmake -G "Ninja" ..
```
Comment (salemgolemugoo) [inline BUILD.md:82]:
```suggestion
cmake -G "Ninja" .. -DOPENSSL_ROOT_DIR=$(brew --prefix)/opt/openssl
```

File: BUILD.md (hunk 1 above)
Comment (Deniskore) [reply BUILD.md:70]: The main reason is the system integration with CMAKE, ports compiled using vcpkg do not affect the current environment and it provides [more convenient way](https://devblogs.microsoft.com/cppblog/vcpkg-using-multiple-enlistments/) to handle multiple versions of libraries.

File: BUILD.md (hunk 3 above)
Comment (Deniskore) [reply BUILD.md:82]: OpenSSL and other libraries will be automatically detected if you use option
`-DCMAKE_TOOLCHAIN_FILE=[vcpkg root]\scripts\buildsystems\vcpkg.cmake`

**Comment by talentlessguy** [comment]
How's PR going?

**Comment by Psyruss77** [comment]
Nice Kult hip

**Comment by EmelyanenkoK** [comment]
Should be moved to [docs/compile](https://github.com/ton-blockchain/docs/blob/master/docs/compile.md).

**Pull request closed without merging**